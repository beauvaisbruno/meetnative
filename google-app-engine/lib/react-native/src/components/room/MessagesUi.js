"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const utilsUi_1 = require("../../utils/utilsUi");
const react_redux_1 = require("react-redux");
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
const react_navigation_1 = require("react-navigation");
const utils_1 = require("../../utils/utils");
const custom_1 = __importDefault(
  require("../../config/native-base-theme/variables/custom")
);
const Loading_1 = __importDefault(require("../../utils/Loading"));
class MessagesUi extends React.Component {
  constructor() {
    super(...arguments);
    this.prevRecipientMsg = null;
  }
  render() {
    const { loadRoom, loadAuthor, loadRecipient, userProfile } = this.props;
    const hasLoad = !loadRoom.fetching && loadAuthor.data && loadRecipient.data;
    if (!hasLoad)
      return React.createElement(
        native_base_1.Content,
        null,
        React.createElement(Loading_1.default, null)
      );
    const messages = utils_1.sortByTimeAsc(
      loadAuthor.data.concat(loadRecipient.data)
    );
    const rows = [];
    const recipientStyle = {
      backgroundColor: custom_1.default.brandSecondaryLight,
      alignSelf: "flex-start"
    };
    const authorStyle = {
      backgroundColor: custom_1.default.brandPrimaryLight,
      alignSelf: "flex-end"
    };
    let lastIsAuthor = null;
    let lastTime = null;
    const contactId = this.props.contactProfile.id;
    messages.forEach(message => {
      try {
        if (contactId !== message.authorId && contactId !== message.recipientId)
          return;
        const isAuthor = message.authorId === userProfile.id;
        if (lastIsAuthor !== null && lastIsAuthor !== isAuthor)
          rows.push(
            React.createElement(native_base_1.View, {
              style: { paddingTop: 10 },
              key: message.time + "-padder"
            })
          );
        if (lastTime === null || message.time - lastTime > 5 * 60 * 1000)
          rows.push(
            React.createElement(
              native_base_1.Text,
              {
                style: {
                  textAlign: "center",
                  color: custom_1.default.textGreyLight
                },
                key: message.time + "-time"
              },
              utilsUi_1.formatDuration(message.time)
            )
          );
        const rowStyle = isAuthor ? authorStyle : recipientStyle;
        rows.push(
          React.createElement(
            native_base_1.Text,
            {
              style: Object.assign(
                {
                  borderRadius: 10,
                  fontSize: utilsUi_1.font(25),
                  color: custom_1.default.textGrey
                },
                utilsUi_1.margin(2),
                utilsUi_1.padding(5),
                rowStyle,
                {
                  minWidth: react_native_responsive_screen_1.widthPercentageToDP(
                    "25%"
                  ),
                  maxWidth: react_native_responsive_screen_1.widthPercentageToDP(
                    "85%"
                  )
                }
              ),
              key: message.time
            },
            message.message
          )
        );
        lastIsAuthor = isAuthor;
        lastTime = message.time;
      } catch (error) {
        console.log("error: ", error);
      }
    });
    return React.createElement(
      native_base_1.Content,
      { ref: ref => (this.scrollView = ref) },
      React.createElement(native_base_1.View, { style: { flex: 1 } }, rows)
    );
  }
  componentDidUpdate() {
    if (this.scrollView) this.scrollView._root.scrollToEnd();
  }
}
exports.MessagesUi = MessagesUi;
exports.mapStateToProps = state =>
  Object.assign({}, state.room, {
    userProfile: state.profile.profileLoad.data
  });
exports.default = react_redux_1.connect(exports.mapStateToProps)(
  react_navigation_1.withNavigation(MessagesUi)
);
//# sourceMappingURL=MessagesUi.js.map
