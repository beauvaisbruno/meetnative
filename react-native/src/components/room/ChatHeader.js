"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const Avatars_1 = require("../profile/img/Avatars");
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
const react_native_auto_height_image_1 = __importDefault(require("react-native-auto-height-image"));
const utils_1 = require("../../utils/utils");
const utilsUi_1 = require("../../utils/utilsUi");
const lang_1 = __importDefault(require("./lang"));
const custom_1 = __importDefault(require("../../config/native-base-theme/variables/custom"));
class ChatHeader extends React.Component {
    constructor(props) {
        super(props);
        this.state = Object.assign({ headerOpen: false }, props.storyState);
    }
    render() {
        const s = utilsUi_1.font(10);
        const m = utilsUi_1.font(15);
        const l = utilsUi_1.font(20);
        const xl = utilsUi_1.font(27);
        const profile = this.props.profile;
        const user = this.props.user;
        return (React.createElement(native_base_1.View, null,
            !this.state.headerOpen && (React.createElement(native_base_1.View, { style: {
                    flexDirection: "row",
                    justifyContent: "space-between",
                    borderBottomWidth: 1,
                    borderBottomColor: custom_1.default.borderColor
                } },
                React.createElement(native_base_1.View, { style: {
                        flexDirection: "row"
                    } },
                    React.createElement(react_native_auto_height_image_1.default, { source: Avatars_1.avatars[profile.avatar], width: react_native_responsive_screen_1.widthPercentageToDP("13%") }),
                    React.createElement(native_base_1.Text, { style: { marginLeft: 5, fontWeight: "bold", fontSize: xl } }, profile.pseudo)),
                React.createElement(native_base_1.Button, { block: true, transparent: true, onPress: () => this.handleSetHeaderVisible(true), testID: "headerVisibleBtn" },
                    React.createElement(native_base_1.Icon, { name: "plus-circle", type: "FontAwesome", fontSize: 10, style: { marginLeft: 0, marginRight: 0, fontSize: 10 } }),
                    React.createElement(native_base_1.Text, { style: { fontSize: s, paddingLeft: 5, paddingRight: 5 } }, lang_1.default("Info"))))),
            this.state.headerOpen && (React.createElement(native_base_1.View, { style: {
                    flexDirection: "row",
                    borderBottomWidth: 1,
                    borderBottomColor: custom_1.default.borderColor
                } },
                React.createElement(react_native_auto_height_image_1.default, { source: Avatars_1.avatars[profile.avatar], width: react_native_responsive_screen_1.widthPercentageToDP("50%") }),
                React.createElement(native_base_1.View, { style: { paddingLeft: 5, width: react_native_responsive_screen_1.widthPercentageToDP("50%") } },
                    React.createElement(native_base_1.View, { style: {
                            flexDirection: "row",
                            justifyContent: "space-between",
                            alignItems: "center"
                        } },
                        React.createElement(native_base_1.View, { style: { flexDirection: "row" } },
                            React.createElement(native_base_1.Text, { style: { fontWeight: "bold", fontSize: l } }, profile.pseudo),
                            React.createElement(native_base_1.Icon, { name: profile.isMale ? "gender-male" : "gender-female", type: "MaterialCommunityIcons", style: {
                                    color: profile.isMale
                                        ? custom_1.default.brandPrimary
                                        : custom_1.default.brandSecondary
                                }, fontSize: l })),
                        React.createElement(native_base_1.Button, { block: true, transparent: true, onPress: () => this.handleSetHeaderVisible(false), testID: "headerVisibleBtn", style: Object.assign({}, utilsUi_1.noMarPad, { height: m + 5 }) },
                            React.createElement(native_base_1.Icon, { name: "close-circle-outline", type: "MaterialCommunityIcons", style: Object.assign({}, utilsUi_1.noMarPad, { fontSize: m, color: custom_1.default.brandDanger }) }))),
                    React.createElement(native_base_1.Text, { style: { color: custom_1.default.brandPrimary, fontSize: m } },
                        profile.city,
                        " ",
                        utils_1.distanceKm(user.lat, user.lon, profile.lat, profile.lon),
                        "km"),
                    React.createElement(native_base_1.Text, { style: { color: custom_1.default.textGreyLight, fontSize: s } },
                        profile.profession,
                        " - ",
                        profile.age + lang_1.default("yo")),
                    React.createElement(native_base_1.Text, { style: { fontWeight: "bold", fontSize: s } }, lang_1.default("Speak")),
                    utilsUi_1.teachRows(user, s, profile.langsToLearn),
                    React.createElement(native_base_1.Text, { style: { fontWeight: "bold", fontSize: s } }, lang_1.default("Learn")),
                    utilsUi_1.learnRows(user, s, profile.langsToTeach))))));
    }
    handleSetHeaderVisible(headerOpen) {
        this.setState({ headerOpen });
    }
}
exports.ChatHeader = ChatHeader;
exports.default = ChatHeader;
