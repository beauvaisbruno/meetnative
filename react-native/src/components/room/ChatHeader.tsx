import { IProfile } from "../../rootReducers";
import * as React from "react";
import { Button, Content, Icon, Text, View } from "native-base";
import { avatars } from "../profile/img/Avatars";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import AutoHeightImage from "react-native-auto-height-image";
import { distanceKm } from "../../utils/utils";
import { font, learnRows, noMarPad, teachRows } from "../../utils/utilsUi";
import lang from "./lang";
import custom from "../../config/native-base-theme/variables/custom";

export type IProps = {
  profile: IProfile;
  user: IProfile;
  storyState?: object;
};
export interface IState {
  headerOpen: boolean;
}

export class ChatHeader extends React.Component<IProps, IState> {
  constructor(props: any) {
    super(props);
    this.state = {
      headerOpen: false,
      ...props.storyState
    };
  }
  render() {
    const s = font(10);
    const m = font(15);
    const l = font(20);
    const xl = font(27);
    const profile = this.props.profile;
    const user = this.props.user;
    return (
      <View>
        {!this.state.headerOpen && (
          <View
            style={{
              flexDirection: "row",
              justifyContent: "space-between",
              borderBottomWidth: 1,
              borderBottomColor: custom.borderColor
            }}
          >
            <View
              style={{
                flexDirection: "row"
              }}
            >
              <AutoHeightImage
                source={avatars[profile.avatar] as number}
                width={wp("13%")}
              />
              <Text style={{ marginLeft: 5, fontWeight: "bold", fontSize: xl }}>
                {profile.pseudo}
              </Text>
            </View>
            <Button
              block
              transparent
              onPress={() => this.handleSetHeaderVisible(true)}
              testID={"headerVisibleBtn"}
            >
              <Icon
                name="plus-circle"
                type="FontAwesome"
                fontSize={10}
                style={{ marginLeft: 0, marginRight: 0, fontSize: 10 }}
              />
              <Text style={{ fontSize: s, paddingLeft: 5, paddingRight: 5 }}>
                {lang("Info")}
              </Text>
            </Button>
          </View>
        )}
        {this.state.headerOpen && (
          <View
            style={{
              flexDirection: "row",
              borderBottomWidth: 1,
              borderBottomColor: custom.borderColor
            }}
          >
            <AutoHeightImage
              source={avatars[profile.avatar] as number}
              width={wp("50%")}
            />
            <View style={{ paddingLeft: 5, width: wp("50%") }}>
              <View
                style={{
                  flexDirection: "row",
                  justifyContent: "space-between",
                  alignItems: "center"
                }}
              >
                <View style={{ flexDirection: "row" }}>
                  <Text style={{ fontWeight: "bold", fontSize: l }}>
                    {profile.pseudo}
                  </Text>
                  <Icon
                    name={profile.isMale ? "gender-male" : "gender-female"}
                    type="MaterialCommunityIcons"
                    style={{
                      color: profile.isMale
                        ? custom.brandPrimary
                        : custom.brandSecondary
                    }}
                    fontSize={l}
                  />
                </View>
                <Button
                  block
                  transparent
                  onPress={() => this.handleSetHeaderVisible(false)}
                  testID={"headerVisibleBtn"}
                  style={{ ...noMarPad, height: m + 5 }}
                >
                  <Icon
                    name="close-circle-outline"
                    type="MaterialCommunityIcons"
                    style={{
                      ...noMarPad,
                      fontSize: m,
                      color: custom.brandDanger
                    }}
                  />
                </Button>
              </View>
              <Text style={{ color: custom.brandPrimary, fontSize: m }}>
                {profile.city}{" "}
                {distanceKm(user.lat!, user.lon!, profile.lat!, profile.lon!)}km
              </Text>
              <Text style={{ color: custom.textGreyLight, fontSize: s }}>
                {profile.profession}
                {" - "}
                {profile.age + lang("yo")}
              </Text>
              <Text style={{ fontWeight: "bold", fontSize: s }}>
                {lang("Speak")}
              </Text>
              {teachRows(user, s, profile.langsToLearn)}
              <Text style={{ fontWeight: "bold", fontSize: s }}>
                {lang("Learn")}
              </Text>
              {learnRows(user, s, profile.langsToTeach)}
            </View>
          </View>
        )}
      </View>
    );
  }
  private handleSetHeaderVisible(headerOpen: boolean) {
    this.setState({ headerOpen });
  }
}

export default ChatHeader;
