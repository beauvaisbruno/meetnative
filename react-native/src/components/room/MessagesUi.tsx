import * as React from "react";
import { Content, Spinner, Text, View } from "native-base";
import { IApplicationState, IProfile } from "../../rootReducers";
import { IMessage, IRoomState, RoomActions } from "./roomReducer";
import { font, formatDuration, margin, padding } from "../../utils/utilsUi";
import { connect } from "react-redux";
import { ScrollView, StyleProp, ViewStyle } from "react-native";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import { NavigationInjectedProps, withNavigation } from "react-navigation";
import { sortByTimeAsc } from "../../utils/utils";
import custom from "../../config/native-base-theme/variables/custom";
import Loading from "../../utils/Loading";

export type IProps = IRoomState &
  NavigationInjectedProps & {
    userProfile: IProfile | undefined;
    contactProfile: IProfile;
  };

export class MessagesUi extends React.Component<IProps> {
  private scrollView: any;
  private prevRecipientMsg: IMessage[] | null = null;

  render() {
    const { loadRoom, loadAuthor, loadRecipient, userProfile } = this.props;
    const hasLoad = !loadRoom.fetching && loadAuthor.data && loadRecipient.data;
    if (!hasLoad)
      return (
        <Content>
          <Loading />
        </Content>
      );
    const messages = sortByTimeAsc(
      loadAuthor.data!.concat(loadRecipient.data!)
    ) as IMessage[];

    const rows: JSX.Element[] = [];
    const recipientStyle: StyleProp<ViewStyle> = {
      backgroundColor: custom.brandSecondaryLight,
      alignSelf: "flex-start"
    };
    const authorStyle: StyleProp<ViewStyle> = {
      backgroundColor: custom.brandPrimaryLight,
      alignSelf: "flex-end"
    };
    let lastIsAuthor: boolean | null = null;
    let lastTime: number | null = null;
    const contactId = this.props.contactProfile.id;
    messages.forEach((message: IMessage) => {
      try {
        if (contactId !== message.authorId && contactId !== message.recipientId)
          return;
        const isAuthor = message.authorId === userProfile!.id;

        if (lastIsAuthor !== null && lastIsAuthor !== isAuthor)
          rows.push(
            <View style={{ paddingTop: 10 }} key={message.time + "-padder"} />
          );
        if (lastTime === null || message.time - lastTime > 5 * 60 * 1000)
          rows.push(
            <Text
              style={{ textAlign: "center", color: custom.textGreyLight }}
              key={message.time + "-time"}
            >
              {formatDuration(message.time)}
            </Text>
          );

        const rowStyle = isAuthor ? authorStyle : recipientStyle;
        rows.push(
          <Text
            style={{
              borderRadius: 10,
              fontSize: font(25),
              color: custom.textGrey,
              ...margin(2),
              ...padding(5),
              ...rowStyle,
              minWidth: wp("25%"),
              maxWidth: wp("85%")
            }}
            key={message.time}
          >
            {message.message}
          </Text>
        );
        lastIsAuthor = isAuthor;
        lastTime = message.time;
      } catch (error) {
        console.log("error: ", error);
      }
    });
    return (
      <Content ref={ref => (this.scrollView = ref)}>
        <View style={{ flex: 1 }}>{rows}</View>
      </Content>
    );
  }
  public componentDidUpdate() {
    if (this.scrollView) this.scrollView!._root.scrollToEnd();
  }
}

export const mapStateToProps = (state: IApplicationState) => ({
  ...state.room,
  userProfile: state.profile.profileLoad.data!
});

export default connect(mapStateToProps)(withNavigation(MessagesUi));
