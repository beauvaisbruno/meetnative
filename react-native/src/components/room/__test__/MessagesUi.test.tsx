import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import * as React from "react";
import ChatHeader from "../ChatHeader";
import MessagesUi from "../MessagesUi";

const initialProps = {
  userProfile: {
    langsToLearn: {
      french: "intermediate",
      italian: "beginner",
      spanish: "fluent"
    },
    age: 23,
    lat: 46.144516,
    pseudo: "pseudo",
    lon: 6.081338,
    langsToTeach: {
      english: "native",
      spanish: "fluent"
    },
    geohash: 3661469831164302,
    city: "Genève",
    avatar: "cute",
    profession: "profession"
  },
  contactProfile: {
    langsToLearn: {
      english: "beginner",
      french: "fluent",
      german: "intermediate"
    },
    profession: "Legal Assistant",
    city: "Genève",
    geohash: 872962206,
    langsToTeach: {
      french: "fluent",
      italian: "bilingual",
      spanish: "native"
    },
    lon: 6.107057129141845,
    pseudo: "Maëlle",
    avatar: "pelican",
    lat: 46.08995392215012,
    age: 23,
    id: "IEY3I5fpxQilwKzPt0Sc"
  },
  loadAuthor: {
    data: [
      {
        message: "other message from user",
        recipient: "IEY3I5fpxQilwKzPt0Sc",
        time: 1551091662000 + 20000,
        author: "C3iSMgBhdmazzcOngiTBpo7USNv2"
      },
      {
        message: "some message from user",
        recipient: "IEY3I5fpxQilwKzPt0Sc",
        time: 1551091662000,
        author: "C3iSMgBhdmazzcOngiTBpo7USNv2"
      }
    ]
  },
  loadRecipient: {
    data: [
      {
        message: "other message to user",
        recipient: "C3iSMgBhdmazzcOngiTBpo7USNv2",
        userIds: "C3iSMgBhdmazzcOngiTBpo7USNv2-IEY3I5fpxQilwKzPt0Sc",
        time: 1551091662000 + 30000,
        author: "IEY3I5fpxQilwKzPt0Sc"
      },
      {
        message: "some message to user",
        recipient: "C3iSMgBhdmazzcOngiTBpo7USNv2",
        userIds: "C3iSMgBhdmazzcOngiTBpo7USNv2-IEY3I5fpxQilwKzPt0Sc",
        time: 1551091662000 + 10000,
        author: "IEY3I5fpxQilwKzPt0Sc"
      }
    ]
  },
  loadRoom: { fetching: false }
};
it("MessagesUi", () => {
  // @ts-ignore
  let tree = shallow(<MessagesUi {...initialProps} />);
  expect(toJson(tree)).toMatchSnapshot();
  tree = shallow(
    // @ts-ignore
    <MessagesUi {...initialProps} loadRoom={{ fetching: true }} />
  );
  expect(toJson(tree)).toMatchSnapshot();
});
