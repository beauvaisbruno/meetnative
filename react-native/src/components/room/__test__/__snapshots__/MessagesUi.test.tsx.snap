// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesUi 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`MessagesUi 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
